Makefile

```diff
@@ -26,13 +27,26 @@ install: lint simplify go-install
 # Reformat and simplify source files.
 simplify:
 ## simplify: Format and simplify the go source code
-	gofmt -w -l -s .
+	@echo "Formating the go source code."
+	@gofmt -w -l -s .
 
 .PHONY: lint
 lint:
 ## lint: run the go linter
-	golangci-lint run -c .golangci.yml
-  
+	@echo "Running go lint"
+	@golangci-lint run -c .golangci.yml
+
+# require terraform to be installed.... HML
+.PHONY: docs
+docs:
+## docs: update the generated terraform docs.
+ifeq ($(shell which terraform && echo true),true)
```

Comment:
That shell line doesn't work, which is probably why you have the line `Unable to generate docs, terraform not installed` in the QA steps. In my experience `command` is usually more reliable than `which`.

```suggestion
HAS_TERRAFORM := $(shell command -v terraform 2> /dev/null)
.PHONY: docs
docs:
## docs: update the generated terraform docs.
ifneq ($(HAS_TERRAFORM),)
```

Reply:
thanks!